### ISSUE TITLE
Logging incompatibility with ActiveRecord 7.0

### ISSUE THREAD

**Issue by bobmaerten** [alekseyl/niceql#20]
Hello,

it seems the abstract `log` method from ActiveRecord changed signature to allow async usage:
[/activerecord-7.0.0/lib/active_record/connection_adapters/abstract_adapter.rb:755](https://github.com/rails/rails/commit/7fc174aadaefc2c0a8a6b7c8a7599dd9ca04811f#diff-460f4e7973c5dd945c51d24df5b0173961190d3645f4e2585fd3003fa1fc0ff7R696)

Resulting in an error when trying to destroy something:
```
> User.destroy_all
  User Load (0.6ms)
SELECT "users".*
  FROM "users"
  TRANSACTION (0.1ms)
BEGIN
  TRANSACTION (0.1ms)
ROLLBACK
ArgumentError: wrong number of arguments (given 6, expected 1..5)
from /Users/bob/.asdf/installs/ruby/3.0.3/lib/ruby/gems/3.0.0/gems/activerecord-7.0.0/lib/active_record/connection_adapters/abstract_adapter.rb:755:in `log'
```

**Comment by bobmaerten** [comment]
I did not push a PR yet, because I don't really know the right way to handle rails version check with gems. Any idea?

**Comment by alekseyl** [comment]
Thanks, for bringing this up. I'll try to close it till the end of big Russia's holidays.

**Comment by alekseyl** [comment]
I think the best idea would be to extract AR integration into a standalone gem and have the set of aligned versions there.

**Comment by alekseyl** [comment]
@bobmaerten, extracted everything to a new integration gem and yet cannot reproduce an issue )))). not in ruby 2.7, not in 3.0.1 and I'm not forcing any arity there, I'm using sql, + *args, may be you have something else added messing with *args structure.

You may try rails_sql_prettifier of a corresponding version ( 7.0.x ), I'm finalizing its first release right now.

**Comment by bobmaerten** [comment]
@alekseyl the problem stills occurs with `prettify_active_record_log_output = true` even with extracted gem.
May be I should open an issue there.

**Comment by alekseyl** [comment]
@bobmaerten, can you provide more info? Or something reproduceable?

**Comment by bobmaerten** [comment]
I try to compile something more actionable tomorrow.

**Comment by bobmaerten** [comment]
Sorry I can't find out what's not working in my environment, more than when I activate `prettify_active_record_log_output = true`, I have an arity error from `/activerecord-7.0.1/lib/active_record/connection_adapters/abstract_adapter.rb:755:in 'log'`.

The only thing that tackle me is when you say using `*args` but from 7.0 there is an optional `async: false` parameter before the block.

7.0+
```ruby
        def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name = nil, async: false, &block)
```

6.1
```ruby
        def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name = nil)
```

**Comment by alekseyl** [comment]
Can you at least print stacktrace?

**Comment by alekseyl** [comment]
Finally I've got it reprodused ))

This is a ruby 2.7 VS 3.0 difference

**Comment by bobmaerten** [comment]
Ok, sorry I forgot to detail my environment. We're running ruby 3.0.1 indeed.

**Comment by alekseyl** [comment]
That was not a problem from the start, my first guess was 100% correct ( inconvenience betwen * and ** args ), but I did a quick test in irb, and that one I've messed up :(((, and got a wrong impression that this is not a problem. 

Sorry I've got stuck during testing, discovered strange behaviour of alias_method, it's not aliasing prepended method instead its aliasing original AbtractLoggerLogger, issue has an easy fix, I can do a quick fix without testing if you needed right now )).

**Comment by bobmaerten** [comment]
No need to rush anything. Take the time to do things right!

**Comment by alekseyl** [comment]
there is the thing I've faced: https://github.com/seattlerb/minitest/pull/891/files 

undef_method  was causing the troubles during test stubbing.. and also one funny stuff blowed my mind: prepending module to a singleton_class. At some point I did that, just to imitate some behaviour on the class, and that also conflicts with the stub.

**Comment by alekseyl** [comment]
fixed in 7.0.3 gem version!
